bbc/simorgh
# WSTEAM1-1444: Adds Chartbeat to Live Pages With Refactor of SPA Functionality

Resolves [WSTEAM1-1444](https://jira.dev.bbc.co.uk/browse/WSTEAM1-1444)

Overall changes
======
- Implementation of adding Chartbeat on Lite pages, as well as refactoring Chartbeat logic, to remove client side routing support.

Code changes
======
- Returns Canonical Chartbeat component when not AMP, catering for lite and canonical
- Adds tests to check that the correct Helmet script tags are added to the page for Lite mode
- Moves relevant Chartbeat code from UserContext to ChartbeatAnalytics component
- Removes code to support SPA/Client side routing for Chartbeat.

Testing
======
Regression should be carried out on canonical, amp and .app, as well as lite testing below:

1. Visit http://localhost:7080/pidgin/articles/cj5l1958ljjo.lite?renderer_env=live
2. Open the Network tab (may need to refresh the page to see all the transferred files)
3. Confirm that `chartbeat.js` is downloaded
4. Stay the page for awhile and observe 'ping' requests to Chartbeat
5. Click on one of the 'ping' requests and check that values are present

Helpful Links
======
OG issue around SPA support for ChartBeat https://github.com/bbc/simorgh/issues/3621

ChartBeat docs around SPA https://docs.chartbeat.com/cbp/tracking/standard-websites/single-page-apps

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)
